docs/proposals/mapper-design.md

```diff
@@ -0,0 +1,120 @@
+---
```

Comment:
So what is this mapper exactly? Is it an application can connect to every devices or framework/SDK that the user need to do some development to connect to differents devices? What role is expected to use this mapper? How this mapper interact with KubeEdge device twin?

Reply:
Mapper is an application that can connect to the devices. We will have multiple mappers like mapper-bluetooth, mapper-zigbee as the definitions and standards used by devices supporting such protocols is standard. For.eg (GATT in bluetooth). There can be custom mappers for devices supporting custom protocols. This is just a generic design for what all components a mapper should have. We plan to provide some sample and generic mappers like for BLE, Zigbee,etc in KubeEdge. User/Developer can have reference these mappers and write their own if their devices are not supported by these samples. We do not plan to impose this design on user/developer but our samples will follow these designs. There maybe some specific things in each individual mapper which are specific to protocol/device are not included here. In general, mapper watches on deviceTwin via the twin topics, if there is any change in the expected state, it is mappers job to perform operations to make actual state of device attribute equal to expected. Mapper will also collect telemetry data from devices (eg: temperature sensor readings) and convert to a format understood by KubeEdge and publish on relevant MQTT topic.

Reply:
OK, so we need to use one concrete protocol as a sample in this design.

Reply:
@sids-b 
What happens to the telemetrydata after it is published to a MQTT topic?
I'm following where the data is going and I'm stuck at scheduler module in bluetooth mapper where it publishes the data to a mqtt topic. I'm not able to find any subscribers to that topic.
Does the telemetry data sent  the cloud ?